### PR TITLE
Announcement - Hack the DEX - Incident Report 4

### PR DESCRIPTION
**Pull request by ryanRfox**
PR to announce Halt Chain by Global Settlement of Short Asset with Supply Greater Than GRAPHENE_MAX_SHARE_SUPPLY

### CODE REVIEW

File: _announcements/2019-01-11-hack-the-dex-report-4.md [hunk 1]
```diff
@@ -0,0 +1,40 @@
+---
+title: Hack the DEX Incident Report - 004
+layout: announcement
+---
+### Halt Chain by Global Settlement of Short Asset with Supply Greater Than GRAPHENE_MAX_SHARE_SUPPLY
+
+On December 6th 2018 the [Hack the DEX](https://hackthedex.io) Team received a report of a potential 
+unhandled exception that would result in halting the blockchain. The report was evaluated, tested, 
+patched and code distributed block producing nodes on December **TODO: XX**, 2018. The BitShares
```
Comment (pmconrad) [inline _announcements/2019-01-11-hack-the-dex-report-4.md]: A "soft fork" was distributed to block producing nodes that prevents the issue from entering the blockchain on Dec 8.
A permanent fix is in the making, see https://github.com/bitshares/bitshares-core/pull/1498 .

File: _announcements/2019-01-11-hack-the-dex-report-4.md [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+title: Hack the DEX Incident Report - 004
+layout: announcement
+---
+### Halt Chain by Global Settlement of Short Asset with Supply Greater Than GRAPHENE_MAX_SHARE_SUPPLY
+
+On December 6th 2018 the [Hack the DEX](https://hackthedex.io) Team received a report of a potential 
+unhandled exception that would result in halting the blockchain. The report was evaluated, tested, 
+patched and code distributed block producing nodes on December **TODO: XX**, 2018. The BitShares 
+network was **not** impacted, **no** user funds were affected and **all operations continue to function**
+as designed.
```
Comment (pmconrad) [inline _announcements/2019-01-11-hack-the-dex-report-4.md]: The soft fork currently prevents certain forms of `call_order_update_operation` from being included in proposals. This restriction will be removed when the permanent fix is activated, i. e. at the time of the next consensus upgrade (aka hardfork).

File: _announcements/2019-01-11-hack-the-dex-report-4.md [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+network was **not** impacted, **no** user funds were affected and **all operations continue to function**
+as designed. 
+
+### Reward
+Special thanks to [@cogutvalera](https://github.com/cogutvalera) for providing the research and 
+prototype code to resolve the bug. The Hack the DEX Team evaluated the submission and awards 
+**TODO: XX,XXX bitUSD** tokens as a reward for disclosing the bug.
+
+### Root Cause
+An unhandled exception existed where, at the time an asset entered global settlement, an asset with a
```
Comment (pmconrad) [inline _announcements/2019-01-11-hack-the-dex-report-4.md]: The bug is triggered when a user force-settles a large amount. Global settlement is not a precondition.

File: _announcements/2019-01-11-hack-the-dex-report-4.md [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+as designed. 
+
+### Reward
+Special thanks to [@cogutvalera](https://github.com/cogutvalera) for providing the research and 
+prototype code to resolve the bug. The Hack the DEX Team evaluated the submission and awards 
+**TODO: XX,XXX bitUSD** tokens as a reward for disclosing the bug.
+
+### Root Cause
+An unhandled exception existed where, at the time an asset entered global settlement, an asset with a 
+supply greater than GRAPHENE_MAX_SHARE_SUPPLY would cause **TODO: a very technical description**.
```
Comment (pmconrad) [inline _announcements/2019-01-11-hack-the-dex-report-4.md]: The execution of a forced settlement happens at a specific time after the user requested it through an `asset_settle_operation`. In very specific cases, this execution would run into an assertion failure, which would prevent any additional blocks from being applied to the chain database, which effectively means the chain is halted.

**Comment by ryanRfox** [comment]
I believe I addressed the review items identified @pmconrad.

File: _announcements/2019-01-11-hack-the-dex-report-4.md [hunk 5]
```diff
@@ -0,0 +1,46 @@
+---
+title: Hack the DEX Incident Report - 004
+layout: announcement
+---
+### Halt Chain by Greater Than MAX_SHARE_SUPPLY
+
+On December 6th 2018 the [Hack the DEX](https://hackthedex.io) Team received a report of a potential 
+unhandled exception that would result in halting the blockchain. The report was evaluated, tested, 
+patched and code distributed block producing nodes on December 8, 2018. The BitShares
```
Comment (jmjatlanta) [inline _announcements/2019-01-11-hack-the-dex-report-4.md]: ''..patched and code **was** distributed **to** block producing...''

**Pull request merged**